tests/copy-index-test.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+#
+# Copyright 2024 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+REKOR_ADDRESS=http://localhost:3000
+REDIS_HOST=localhost
+REDIS_PORT=6379
+REDIS_PASSWORD=test
+MYSQL_HOST=127.0.0.1
+MYSQL_PORT=3306
+MYSQL_USER=test
+MYSQL_PASSWORD=zaphod
+MYSQL_DB=test
+
+testdir=$(mktemp -d)
+
+declare -A expected_artifacts
+declare -A expected_keys
+
+source $(dirname "$0")/index-test-utils.sh
+
+trap cleanup EXIT
+
+make_intoto_entries() {
+    set -e
+    for v in 0.0.1 0.0.2 ; do
+        rekor-cli --rekor_server $REKOR_ADDRESS upload \
+            --type intoto:${v} \
```

Comment:
Can we also add a test for the `dsse` type?

Reply:
done